lambda.py

```diff
@@ -18,38 +20,59 @@
 
 
 def lambda_handler(event, context):
+    print('Processing a new event...')
+    print('Event: {}. Context: {}'.format(event, context))
+
     bucket_name = event['Records'][0]['s3']['bucket']['name']
-    bucket = s3.Bucket(bucket_name)
     key = urllib.parse.unquote_plus(event['Records'][0]['s3']['object']['key'], encoding='utf-8')
+    print('Processing key "{}" from bucket "{}"...'.format(key, bucket_name))
+
+    bucket = s3.Bucket(bucket_name)
+
     try:
         folder = get_folder_key(key)
-        folder_content_keys = list_content_in_subfolders(bucket, folder)
+        print('Extracted folder "{}" from key'.format(folder))
+        folder_content_keys = list_pom_files_in_subfolders(bucket, folder)
+        print('Found .pom files: {}'.format(folder_content_keys))
         metadata = generate_release_maven_metadata(folder_content_keys)
+        print('Generated maven-metadata content: {}'.format(metadata))
         upload_s3_file(bucket_name, folder, METADATA_BASE_FILE_NAME, metadata, content_type='text/xml')
+        print('Uploaded new maven-metadata.xml')
 
         checksums = generate_checksums(metadata)
-        print(checksums)
+        print('New maven-metadata.xml checksums: {}'.format(checksums))
         for type_, sum_ in checksums.items():
             upload_s3_file(bucket_name, folder, '{}.{}'.format(METADATA_BASE_FILE_NAME, type_), sum_)
+            print('Uploaded new {} checksum file'.format(type_))
 
     except Exception as e:
         print(e)
-        raise e
+        raise
+
+    print('Done processing folder "{}"'.format(folder))
 
 
 def get_folder_key(key):
-    return '/'.join(key.split('/')[:-2])
+    split_path_without_last_two_items = key.split('/')[:-2]
+    folder = '/'.join(split_path_without_last_two_items)
+    folder_with_trailing_slash = '{}/'.format(folder)
+    return folder_with_trailing_slash
 
 
-def list_content_in_subfolders(bucket, folder_key):
+def list_pom_files_in_subfolders(bucket, folder_key):
     return [
-        file.key for file in bucket.objects.all()
-        if file.key.startswith(folder_key) and '/' in file.key.strip('{}/'.format(folder_key))
+        file.key for file in bucket.objects.filter(Prefix=folder_key)
+        if file.key.endswith('.pom')
     ]
 
 
 def get_group_id(key):
-    return '.'.join(key.split('/')[:-3])
+    folder_names_but_last_three = key.split('/')[:-3]
```

Comment:
Nit (not blocking, just an optional opinion) I'm not a huge fan of splitting lines with trailing `\` so I'm usually leaning for less python solutions in cases like this. Rewriting to following seems easier to read but, as said, not blocking and a totally personal preference:

```
if folder_names_but_last_three[0] in ('maven', 'maven2'):
   # an explanatory comment maybe?
   folder_names_without_maven_prefix = folder_names_but_last_three[1:]
else
   # an explanatory comment maybe?
   folder_names_without_maven_prefix = folder_names_but_last_three
```

Reply:
Done!

Reply:
Sweet!